tfhe/src/integer/server_key/comparator.rs

```diff
@@ -562,104 +660,131 @@ impl<'a> Comparator<'a> {
+                // We have to handle both part of the work
+                // And the sign bit is located in the most_significant_blocks
+                let (lsb_sign, msb_sign) = rayon::join(
                     || {
-                        let msb_cmp_zero = self.server_key.compare_blocks_with_zero(
-                            &most_significant_blocks[..most_significant_blocks.len() - 1],
-                            ZeroComparisonType::Equality,
+                        let lsb_signs = self.unchecked_scalar_block_slice_compare_parallelized(
+                            least_significant_blocks,
+                            &scalar_blocks[..least_significant_blocks.len()],
                         );
-                        let are_all_msb_equal_to_zero =
-                            self.server_key.are_all_comparisons_block_true(msb_cmp_zero);
-                        let lut = self.server_key.key.generate_lookup_table(|x| {
-                            if x == 1 {
-                                Self::IS_EQUAL
-                            } else {
-                                Self::IS_SUPERIOR
-                            }
-                        });
-                        self.server_key
-                            .key
-                            .apply_lookup_table(&are_all_msb_equal_to_zero, &lut)
+                        self.reduce_signs_parallelized(lsb_signs, |x| x)
                     },
                     || {
+                        // most_significant_blocks has the sign block, which must be handled
+                        let are_all_msb_equal_to_zero = self.server_key.are_all_blocks_zero(
+                            &most_significant_blocks[..most_significant_blocks.len() - 1],
+                        );
                         let sign_bit_pos = self.server_key.key.message_modulus.0.ilog2() - 1;
-                        let lut2 = self.server_key.key.generate_lookup_table(|x| {
-                            let x = x % self.server_key.key.message_modulus.0 as u64;
-                            let sign_bit_is_set = (x >> sign_bit_pos) == 1;
-                            if sign_bit_is_set {
-                                Self::IS_INFERIOR
-                            } else if x != 0 {
-                                Self::IS_SUPERIOR
-                            } else {
-                                Self::IS_EQUAL
-                            }
-                        });
-                        self.server_key.key.apply_lookup_table(
-                            &most_significant_blocks[most_significant_blocks.len() - 1],
-                            &lut2,
+
+                        let lut = self.server_key.key.generate_lookup_table_bivariate(
+                            |sign_block, msb_are_zeros| {
+                                let sign_bit_is_set = (sign_block >> sign_bit_pos) == 1;
+                                let sign_block_ordering = if sign_bit_is_set {
```

Comment:
some comments could be welcome here for the if block, same a bit later for similar if blocks